logstash-core/lib/logstash/util/wrapped_acked_queue.rb

```diff
@@ -168,24 +157,14 @@ def read_batch
 
       def start_metrics(batch)
         thread = Thread.current
-        @mutex.lock
-        begin
-          @inflight_batches[thread] = batch
-        ensure
-          @mutex.unlock
-        end
+        @inflight_batches[thread] = batch
         @inflight_clocks[thread] = java.lang.System.nano_time
       end
 
       def close_batch(batch)
         thread = Thread.current
-        @mutex.lock
-        begin
-          batch.close
-          @inflight_batches.delete(thread)
-        ensure
-          @mutex.unlock
-        end
+        batch.close
```

Comment:
I can't come up with a scenario where this would break but could we have a race bug here by not wrapping both instructions into a lock?

Reply:
@jsvd logically impossible I think because `batch.close` doesn't reference the state of `inflight_batches` and neither is there any code that references the fact `close` was called on the `batch` in relation to modifying `inflight_batches` => I think we're safe + we do the same in the other queue anyways and haven't seen issues with that in months :)